### PR TITLE
fix: forward given role to chip component

### LINKED ISSUE #8353: Bug - Chip - Chip component does not accept user given attributes
**Describe the problem**
I'm trying to give a role attribute to the chip component but it is not applying to the element

**Expected behavior**
It should take addition attributes and apply to the element

**Is this issue blocking you?**
It's not a blocker

**What is your environment?**
 - OS: RHEL
 - Browser: Chrome
 - Version 106.0.5249.119

**What is your product and what release date are you targeting?**
Red Hat Customer Dashboard

### PR DESCRIPTION
**Pull request by erkanercan**
I wasn't sure if we needed to forward all props like `{...this.props}`, so I only added the role into the component.

**What**: Closes #8353

**Additional issues**: -

### CODE REVIEW

**Comment by nicolethoen** [comment]
> I wasn't sure if we needed to forward all props like `{...this.props}`

This is a good question. I feel as though we usually do want to spread the props - but the props interface extends HTMLDivElement and the chip allows for customization of the chip component. I think changing the interface definition is a breaking change, but maybe something we should do. WDYT @wise-king-sullyman ?

**Comment by wise-king-sullyman** [comment]
@nicolethoen I'm generally in favor of spreading props on any component that we export, and it looks like we spread props to customizable components in at least a few places already (Card, Helper Text, and Page Section as examples) so I agree that it would make sense to do the same here.

Regarding needing a breaking change in the interface I'm honestly not sure. The Page Section interface extends HTMLDivElement as well, and that doesn't seem to cause any type errors 🤔

**Comment by nicolethoen** [comment]
@erkanercan Rather than passing only the role, which is not a declared prop of the Chip, let's spread all props and see if any type issues get raised.

**Comment by erkanercan** [comment]
I spread the props, can you retake a look at it? @nicolethoen

File: packages/react-core/src/components/Chip/Chip.tsx [hunk 1]
```diff
@@ -88,6 +88,7 @@ export class Chip extends React.Component<ChipProps, ChipState> {
     return (
       <Component
         onClick={onClick}
+        {...this.props}
```
Comment (nicolethoen) [inline packages/react-core/src/components/Chip/Chip.tsx:91]: let's spread the props at the end of the component. That way if the consumer wants to override the `type` or other attributes of the component, they can.

File: packages/react-core/src/components/Chip/Chip.tsx [hunk 2]
```diff
@@ -106,6 +107,7 @@ export class Chip extends React.Component<ChipProps, ChipState> {
     const Component = component as any;
     return (
       <Component
+        {...this.props}
```
Comment (nicolethoen) [inline packages/react-core/src/components/Chip/Chip.tsx:110]: same here, we should spread the props at the end

**Review by nicolethoen — CHANGES_REQUESTED** [review]

**Comment by nicolethoen** [comment]
@erkanercan let me know if you're able to make these updates. If not, I'm happy to help move it along

**Comment by erkanercan** [comment]
Hi again @nicolethoen, sorry for the late answer.
I've spread the props at the end like you mentioned, please let me know if any changes are needed and I'll do them ASAP.

**Comment by erkanercan** [comment]
Hmm, it seems like when we spread the props at the end, it overrides some of the props. 
I guess we cannot accept this happening, any idea? Do you want me to give the class names manually(I also guess that is not the right method)? @nicolethoen

**Comment by nicolethoen** [comment]
... hm. It's possible we need to pull out some of the props and set them more intentionally if things are getting overwritten.

oh! i think I see it. On lines 86 and 98 such as in this snippet
```
const { children, className, onClick, closeBtnAriaLabel, isReadOnly, component, ouiaId } = this.props;
```
I think we want to actually update it to the following:
```
const { children, className, onClick, closeBtnAriaLabel, isReadOnly, component, ouiaId, ...props } = this.props;
```
Then, when we spread the props, we will want to spread them with `...props` rather than `...this.props`. This will make sure that we are not overriding any of the props that you've pulled out, such as `children`, `className`, etc...

**Comment by erkanercan** [comment]
> 

I didn't know about it, thanks for pointing it out! I refactored the code as you mentioned.

**Review by nicolethoen — APPROVED** [review]

**Review by tlabaj — APPROVED** [review]

**Pull request merged**